nym-vpn-core/crates/nym-vpn-account-controller/src/ecash_client.rs

```diff
@@ -23,15 +28,34 @@ impl VpnEcashApiClient {
         })
     }
 
-    pub(crate) async fn get_master_verification_key(
+    async fn _get_master_verification_key(
         &self,
     ) -> Result<MasterVerificationKeyResponse, VpnEcashApiClientError> {
         self.inner
             .get_json(&["/v1", "/ecash", "/master-verification-key"], NO_PARAMS)
             .await
     }
 
-    pub(crate) async fn get_aggregated_coin_indices_signatures(
+    pub(crate) async fn get_master_verification_key(&self) -> Result<EpochVerificationKey, Error> {
+        let master_verification_key = self
+            ._get_master_verification_key()
+            .await
+            .map_err(Error::from)?
+            .key;
+
+        // Temporary workaround
+        let current_epoch = self
+            ._get_aggregated_coin_indices_signatures()
```

Comment:
that seems quite nasty as coin indices are the biggest (size wise)

Reply:
can't you perhaps combine queries for the verification key with coin index signatures? because you only need to get both of them once, when DKG epoch changes (i.e. at the same time)

Reply:
This stuff will change once we the credential-proxy is integrated with the nym-vpn-api. Then I wont need to reach out directly to any ecash endpoints on nym-api

But yes, the idea is to only download them once an epoch if possible